stdlib/builtins.ncl

```diff
@@ -1,90 +1,90 @@
 {
   builtins = {
-    isNum : Dyn -> Bool
+    is_num : Dyn -> Bool
     | doc m#"
       Checks if the given value is a number.
 
       For example:
       ```nickel
-        isNum 1 =>
+        is_num 1 =>
           true
-        isNum "Hello, World!" =>
+        is_num "Hello, World!" =>
           false
       ```
       "#m
-    = fun x => %isNum% x,
+    = fun x => %is_num% x,
 
-    isBool : Dyn -> Bool
+    is_bool : Dyn -> Bool
     | doc m#"
       Checks if the given value is a boolean.
 
       For example:
       ```nickel
-        isBool false =>
+        is_bool false =>
           true
-        isBool 42 =>
+        is_bool 42 =>
           false
       ```
       "#m
-    = fun x => %isBool% x,
+    = fun x => %is_bool% x,
 
-    isStr : Dyn -> Bool
+    is_str : Dyn -> Bool
     | doc m#"
       Checks if the given value is a string.
 
       For example:
       ```nickel
-        isBool true =>
+        is_bool true =>
           false
-        isBool "Hello, World!" =>
+        is_bool "Hello, World!" =>
           true
       ```
       "#m
-    = fun x => %isStr% x,
+    = fun x => %is_str% x,
 
-    isFun : Dyn -> Bool
+    is_fun : Dyn -> Bool
     | doc m#"
       Checks if the given value is a function.
 
       For example
       ```nickel
-        isFun (fun x => x) =>
+        is_fun (fun x => x) =>
           true
-        isFun 42 =>
+        is_fun 42 =>
           false
       ```
       "#m
-    = fun x => %isFun% x,
+    = fun x => %is_fun% x,
 
-    isList : Dyn -> Bool
+    is_list : Dyn -> Bool
     | doc m#"
       Checks if the given value is a list.
 
       For example
       ```nickel
-        isList [ 1, 2 ] =>
+        is_list [ 1, 2 ] =>
           true
-        isList 42 =>
+        is_list 42 =>
           false
       ```
       "#m
-    = fun x => %isList% x,
+    = fun x => %is_list% x,
 
-    isRecord : Dyn -> Bool
+    is_record : Dyn -> Bool
     | doc m#"
       Checks if the given value is a record.
 
       For example
       ```nickel
-        isRecord [ 1, 2 ] =>
+        is_record [ 1, 2 ] =>
           false
-        isRecord { hello = "Hello", world = "World" } =>
+        is_record { hello = "Hello", world = "World" } =>
           true
       ```
       "#m
-    = fun x => %isRecord% x,
+    = fun x => %is_record% x,
 
-    typeOf : Dyn -> <
+    typeof : Dyn -> <
```

Comment:
Should we really follow C/C++ precedent here, though? For example, [this random Rust crate](https://docs.rs/typed/1.0.0/typed/fn.type_of.html) uses `type_of`, which IMHO is more consistent with our and Rust's naming convention.

Reply:
The value of any `foo` applied to `baz` is the `foo` _of_ `baz`. So any `_of` in a function is redundant and shouldn't be part of any naming convention. The only reason something like it is there in C/C++ and elsewhere is presumably because `type` is a reserved keyword. I saw the "random Rust crate" you mention, but with 480 downloads over its lifetime, it hardly has enough clout to set a meaningful precedent.

Reply:
Ok, fair enough. By the way `type` is currently not a reserved keyword in Nickel, and we've been trying to avoid to do so when possible (such as with type aliases) as it may easily appear in configurations and would be annoying to escape everywhere (e.g. in the current schema of NixOS modules). However it is not be good idea to tie our hands now and prevent any future use of a `type` keyword, so let's go with `typeof`.

Reply:
Thinking about this more: in C/C++, typeof is an operator. Probably has that name because no other keyword contains an underscore. In Python, the equivalent is a function called `type()`. Could we do the same here? `type` doesn't appear to be a reserved keyword yet and maybe we never want it to be. I have provisionally updated the PR accordingly.

Reply:
Hm, don't know what to make of the message timestamps above. You somehow answered my question from 24 minutes in the future. :)

I've kept `typeof` then. It's easier to go `typeof` -> `type` later, than the other way around.

Reply:
I actually answered from the future indeed :stuck_out_tongue:  I wrote my reply as an answer to your first comment without seeing the following ones.